protocol/x/listing/keeper/listing.go

```diff
@@ -187,3 +196,70 @@ func (k Keeper) GetListingVaultDepositParams(
 	k.cdc.MustUnmarshal(b, &vaultDepositParams)
 	return vaultDepositParams
 }
+
+// Function to deposit to the megavault for a new PML market
+// This function deposits money to the megavault, transfers the new vault
+// deposit amount to the new market vault and locks the shares for the deposit
+func (k Keeper) DepositToMegavaultforPML(
+	ctx sdk.Context,
+	fromSubaccount satypes.SubaccountId,
+	clobPairId uint32,
+) error {
+	// Get the listing vault deposit params
+	vaultDepositParams := k.GetListingVaultDepositParams(ctx)
+
+	// Deposit to the megavault
+	totalDepositAmount := big.NewInt(0)
+	totalDepositAmount.Add(
+		vaultDepositParams.NewVaultDepositAmount.BigInt(),
+		vaultDepositParams.MainVaultDepositAmount.BigInt(),
+	)
+	mintedShares, err := k.VaultKeeper.DepositToMegavault(
+		ctx,
+		fromSubaccount,
+		totalDepositAmount,
+	)
+	if err != nil {
+		return err
+	}
+
+	vaultId := vaulttypes.VaultId{
+		Type:   vaulttypes.VaultType_VAULT_TYPE_CLOB,
+		Number: clobPairId,
+	}
+
+	// Transfer the new vault deposit amount to the new market vault
+	err = k.VaultKeeper.AllocateToVault(
+		ctx,
+		vaultId,
+		vaultDepositParams.NewVaultDepositAmount.BigInt(),
+	)
+	if err != nil {
+		return err
+	}
+
+	// Lock the shares for the new vault deposit amount
+	err = k.VaultKeeper.LockShares(
+		ctx,
+		fromSubaccount.Owner,
+		vaulttypes.NumShares{NumShares: dtypes.NewIntFromBigInt(mintedShares)},
+		uint32(ctx.BlockHeight())+vaultDepositParams.NumBlocksToLockShares,
+	)
+	if err != nil {
+		return err
+	}
+
+	// Activate vault to quoting status
+	err = k.VaultKeeper.SetVaultParams(
```

Comment:
to be more future-proof, maybe first fetch existing vault params:
```
k.VaultKeeper.GetVaultParams(...)
```

Reply:
what are we future-proofing here?

Reply:
if we add more things to `VaultParams` or if `quotingParams` is not initialized to nil